### ISSUE TITLE
Limited Q Value Range

### ISSUE THREAD

**Issue by jaakkopasanen** [jaakkopasanen/AutoEq#253]
Parametric equalizer optimizer produces extreme values for Q in some occasions. For example this [DT 770](https://github.com/jaakkopasanen/AutoEq/tree/ce1a3d61b5f9f1153edae21cea1a871345722725/results/crinacle/gras_43ag-7_harman_over-ear_2018/Beyerdynamic%20DT%20770%2080%20Ohm) result has Q of 0, while it's in practice 0.004, it's still too small value for many equalizer UIs.

Limit the Q value range to some reasonable range like 0.1 - 10.0.

**Comment by JaKasb** [comment]
#### I suggest to include the Q-Values into the Loss-Function for the Tensorflow Optimizer.
https://github.com/jaakkopasanen/AutoEq/blob/aad110b2ae899a08b08e6b9385e81e2d640d0015/frequency_response.py#L523
In machine learning this would be called Regularization.
However Q-Values behave differently than Multiplication-Weigths for Matrix-Multiplication.
0 < Q <= 1/2 should be safe.
See https://en.wikipedia.org/wiki/Q_factor#Q_factor_and_damping.

I see 2 problems for Regularization on Q:
The Regularization-Function should penalize oscillating/ringing/overshooting filters but should allow damped filters.
The required Regularization-Function should quantify the overshoot in the time domain.
This function must be differentiable for the optimizer.
Maybe a function in https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/nn could be adapted as regularization-function.

The second problem is that Regularization requires a tuning parameter lambda:
`loss = tf.reduce_mean(tf.square(eq_op - eq_target))  + lambda*some_function(q_array)`
Lambda is a tune-able Hyperparameter.
In the end, one must decide for a tradeoff between Frequency Error and Ringing-Error.

#### Maybe there is another way for implementing the Regularization on Q ?
(My knowledge about Tensorflow is very basic. I've never used it)
https://github.com/jaakkopasanen/AutoEq/blob/aad110b2ae899a08b08e6b9385e81e2d640d0015/frequency_response.py#L424
`tf.get_variable('Q', initializer=Q_init, dtype='float32', regularizer='???', constraint='???')`
https://www.tensorflow.org/api_docs/python/tf/compat/v1/get_variable

#### Maybe Include the the Maximum-Slope into the Loss-Function ?
The resulting equalizer is the sum of the individual biquads.
Therefore the equalizer transfer function should be differentiable.
I assume the optimizer can un-tangle the filters and optimize the max-slope.
Again, the optimizer requires a differentiable function on the max-slope.

#### Time Domain Analysis
The Overshoot = Ringing in the time-domain, can be measured by convolving a Unit-Step-Function with the filter.
However I doubt that the convolution is differentiable and accessible for the optimizer.
Maybe the tensorflow-convolution can be usefull ?
After all, tensorflow supports Convolutional Neuronal Networks.

**Comment by bhack** [comment]
Another interesting approach to explore could be:
https://dafx2020.mdw.ac.at/proceedings/papers/DAFx2020_paper_7.pdf
